### PR TITLE
Added api to show code

### PR DESCRIPTION
**Pull request by atirabagwan**

### CODE REVIEW

File: app/controllers/api/v1/drives_candidates_controller.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 module Api
   module V1
     class DrivesCandidatesController < ApiController
+      before_action :find_submission
```
Comment (anilmaurya) [inline app/controllers/api/v1/drives_candidates_controller.rb:6]: `before_action` will be applicable to other action as well which is not required.

File: app/controllers/api/v1/drives_candidates_controller.rb [hunk 2]
```diff
@@ -14,6 +15,21 @@ def update
...
+          render_success(data: { answer: @submission.answer }, message: I18n.t('success.message'))
+        else
+          render_error(message: I18n.t('not_found.message'))
+        end
+      end
+
+      private
+
+      def find_submission
+        @drive_candidate = DrivesCandidate.find(params[:drives_candidate_id])
```
Comment (anilmaurya) [inline app/controllers/api/v1/drives_candidates_controller.rb:30]: What if wrong drives_candidate_id is provided

File: app/controllers/api/v1/drives_candidates_controller.rb [hunk 3]
```diff
@@ -14,6 +15,21 @@ def update
...
+        @submission = Submission.find_by(drives_candidate_id: @drive_candidate.id)
```
Comment (anilmaurya) [inline app/controllers/api/v1/drives_candidates_controller.rb:31]: We need to consider submission with highest marks, not the first one

File: spec/controllers/api/v1/drives_candidate_spec.rb [hunk 4]
```diff
@@ -31,4 +33,23 @@
       end
     end
   end
+
+  describe 'Get # show_code' do
+    context 'with valid params' do
+      it 'retuns code' do
+        get :show_code, params: { drives_candidate_id: @drives_candidate.id }
+        result = json
+        expect(result['data']['answer']).to eq(@submission.answer)
```
Comment (anilmaurya) [inline spec/controllers/api/v1/drives_candidate_spec.rb:42]: Add spec with following scenario
1. Candidate has done multiple submissions
2. Candidate has done only one submission
3. Candidate has not done any submission

File: app/controllers/api/v1/drives_candidates_controller.rb (hunk 2 above)
Comment (atirabagwan) [reply app/controllers/api/v1/drives_candidates_controller.rb:30]: it returns Record Not Found.

**Pull request merged**